packet/flatcar-linux/kubernetes/workers/cl/install.yaml.tmpl

```diff
@@ -64,44 +63,20 @@ storage:
             # Ignore the disk on which Linux is installed when selecting disks for RAID.
             local osdisk="$1"
             local major_numbers="$2"
-
-            # Select disks for RAID.
-            local disks=$(lsblk -lnpd -I "$${major_numbers}" \
-              | sort -h -k 4,4 \
-              | grep -vE "^$${osdisk} " \
-              | awk '{x=$1 " " x;} END{print x}'
-            )
-            local count=$(echo "$$disks" | wc -w)
-
-            # Exit if we don't have any disks to create an array
-            [ $$count -lt 1 ] && return 0
-
-            # Create, format and mount array.
-            local extra_opts=""
-            if [ $$count -lt 2 ]; then
-              # Force array creation even with one disk
-              extra_opts="--force"
-            fi
-
-            mdadm --create /dev/md/node-local-storage --homehost=any $$extra_opts --verbose --name=node-local-storage --level=0 --raid-devices="$${count}" $${disks}
-            cat /proc/mdstat
-            mkfs.ext4 /dev/md/node-local-storage
-          }
-
-          function create_disk_specific_data_raid() {
-            # Ignore the disk on which Linux is installed when selecting disks for RAID.
-            local osdisk="$1"
-            local major_numbers="$2"
             # for hdd value is 1 and for ssd value is 0
             local disk_type="$3"
             # RAID device path on disk
             local device_path="$4"
             local setup_fs_on_raid="$5"
 
-            # Select disks for RAID.
-            local disks=$(lsblk -lnpd -o name,rota -I "$${major_numbers}" \
-              | grep "$${disk_type}" \
-              | sort -h -k 4,4 \
+            if [ $$disk_type -lt 0 ]; then
+              local d=$(lsblk -lnpd -I "$${major_numbers}" -x size)
+            else
+              local d=$(lsblk -lnpd -o name,rota -I "$${major_numbers}" -x size \
+                | grep "$${disk_type}"
+              )
+            fi
+            local disks=$(echo "$$d" \
               | grep -vE "^$${osdisk} " \
```

Comment:
Excluding the OS disk like that works but in general, if we are not in the installer image, there may be other disks which are used already and should be excluded. So when this is ported to be run on the first boot of a Packet Flatcar Linux instance, disks with mountpoints should be excluded so that it only takes unused disks (similar to that: https://github.com/kinvolk/lokomotive-kubernetes/pull/28/commits/a2aa822197a02941dfee12f2edde07a8afd434b0#diff-b6c7caf796cd86bdcdd936319b1793a1R148).